### PR TITLE
Setup wizard added for new installations

### PR DESCRIPTION
**Pull request by zurdi15**
Setup wizard added for new installations:

 - Create default admin user
 - Check metadata providers are correctly configured

Depends on:
- #964

### CODE REVIEW

File: backend/endpoints/heartbeat.py [hunk 1]
```diff
@@ -27,6 +28,7 @@ def heartbeat() -> HeartbeatResponse:
 
     return {
         "VERSION": get_version(),
+        "SETUP_WIZARD": len(db_user_handler.get_admin_users()) == 0,
```
Comment (gantoine) [inline backend/endpoints/heartbeat.py]:
```suggestion
        "SHOW_SETUP_WIZARD": len(db_user_handler.get_admin_users()) == 0,
```

File: backend/endpoints/user.py [hunk 2]
```diff
@@ -16,7 +16,10 @@
 
 
 @protected_route(
-    router.post, "/users", ["users.write"], status_code=status.HTTP_201_CREATED
+    router.post,
+    "/users",
+    [] if len(db_user_handler.get_admin_users()) == 0 else ["users.write"],
```
Comment (gantoine) [inline backend/endpoints/user.py]: 👍🏼

File: backend/handler/database/users_handler.py [hunk 3]
```diff
@@ -42,4 +43,10 @@ def delete_user(self, id: int, session: Session = None):
...
     @begin_session
     def get_admin_users(self, session: Session = None) -> list[User]:
-        return session.scalars(select(User).filter_by(role=Role.ADMIN)).all()
+        try:
+            return session.scalars(select(User).filter_by(role=Role.ADMIN)).all()
+        except ProgrammingError as e:
+            if "Table" in str(e.orig) and "doesn't exist" in str(e.orig):
+                return []
+            else:
+                raise ProgrammingError from e
```
Comment (gantoine) [inline backend/handler/database/users_handler.py]: Can you add a comment explaining this error block?

File: frontend/src/services/api/index.ts [hunk 4]
```diff
@@ -37,9 +38,12 @@ api.interceptors.response.use(
   },
   (error) => {
     if (error.response?.status === 403) {
+      const allCookies = Cookies.get(); // Get all cookies
+      for (const cookie in allCookies) {
+        Cookies.remove(cookie); // Remove each cookie
+      }
       router.push({
         name: "login",
-        params: { next: router.currentRoute.value.path },
```
Comment (gantoine) [inline frontend/src/services/api/index.ts:42]: why remove this? it's what brings users back to the page they were just on after login

File: frontend/src/views/Login.vue [hunk 5]
```diff
@@ -1,29 +1,26 @@
...
 const username = ref("");
 const password = ref("");
 const visiblePassword = ref(false);
 const logging = ref(false);
 
-function login() {
+async function login() {
   logging.value = true;
-  identityApi
+  await identityApi
```
Comment (gantoine) [inline frontend/src/views/Login.vue:20]: this form of using await and then is odd, don't worry about fixing it in this PR, i want to refactor those anyways

File: frontend/src/views/Setup.vue [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+    name: "MobyGames",
+    value: "moby",
+    logo_path: "/assets/scrappers/moby.png",
+    disabled: !heartbeat.value.METADATA_SOURCES?.MOBY_API_ENABLED,
+  },
+  {
+    name: "SteamgridDB",
+    value: "sgdb",
+    logo_path: "/assets/scrappers/sgdb.png",
+    disabled: !heartbeat.value.STEAMGRIDDB_ENABLED,
```
Comment (gantoine) [inline frontend/src/views/Setup.vue]: should also be under `METADATA_SOURCES`

```suggestion
    disabled: !heartbeat.value.METADATA_SOURCES?.STEAMGRIDDB_ENABLED,
```

**Review by gantoine — CHANGES_REQUESTED** [review]

File: backend/handler/database/users_handler.py (hunk 3 above)
Comment (zurdi15) [reply backend/handler/database/users_handler.py]: hmm i can't remember why i added that there, maybe something in the middle of the development, i'll remove it

File: frontend/src/views/Login.vue (hunk 5 above)
Comment (zurdi15) [reply frontend/src/views/Login.vue:20]: yeah sounds good, we need to revisit the whole users management system anyways!

File: frontend/src/services/api/index.ts (hunk 4 above)
Comment (zurdi15) [reply frontend/src/services/api/index.ts:42]: it was not working for me and tried to fix it but forgot about it. Fixing it now!

**Review by gantoine — APPROVED** [review]

**Pull request merged**